File: src/InferenceEngine/cpp/model/detection.cpp
#include "detection.hpp"

Detection::Detection(const int classId, const float score, int *locations) : classId(classId), score(score)
{
    this->bbox.xMin = locations[0];
    this->bbox.yMin = locations[1];
    this->bbox.xMax = locations[2];
    this->bbox.yMax = locations[3];
}

int Detection::getClassId(void) { return this->classId; }
float Detection::getScore(void) { return this->score; }
BoundingBox Detection::getBoundingBox(void) { return this->bbox; }

File: src/InferenceEngine/python/model/detection.py
from dataclasses import dataclass

@dataclass
class BoundingBox:
    xMin: int
    yMin: int
    xMax: int
    yMax: int

class Detection:
    def __init__(self, class_id: int, score: float, locations: list[int]):
        self.class_id = class_id
        self.score = score
        self.bbox = BoundingBox(
            xMin=int(locations[0]),
            yMin=int(locations[1]),
            xMax=int(locations[2]),
            yMax=int(locations[3])
        )

    def get_class_id(self) -> int:
        return self.class_id

    def get_score(self) -> float:
        return self.score

    def get_bounding_box(self) -> BoundingBox:
        return self.bbox